app/services/update_status_service.rb

```diff
@@ -137,4 +151,8 @@ def create_previous_edit!
   def create_edit!
     @status.snapshot!(account_id: @account_id)
   end
+
+  def significant_changes?
+    @status.changed? || @poll_changed || @media_attachments_changed || (@previous_expires_at != @status.preloadable_poll&.expires_at)
```

Comment:
I'd move the `@previous_expires_at != @status.preloadable_poll&.expires_at` in the poll creation/edition code and merge it into `@poll_changed`, but it looks sound.